cmd/dra-example-controller/driver.go

```diff
@@ -244,18 +246,20 @@ func (d driver) unsuitableNode(ctx context.Context, pod *corev1.Pod, allcas []*c
 
 	perKindCas := make(map[string][]*controller.ClaimAllocation)
 	for _, ca := range allcas {
-		var kind string
-		switch ca.ClaimParameters.(type) {
-		case *gpucrd.GpuClaimParametersSpec:
-			kind = gpucrd.GpuClaimParametersKind
+		switch ca.Claim.Spec.ParametersRef.Kind {
+		case gpucrd.GpuClaimParametersKind:
+			perKindCas[gpucrd.GpuClaimParametersKind] = append(perKindCas[gpucrd.GpuClaimParametersKind], ca)
+		default:
+			return fmt.Errorf("unsupported ResourceClaimParameters kind: %+v", ca.Claim.Spec.ParametersRef.Kind)
 		}
-		perKindCas[kind] = append(perKindCas[kind], ca)
 	}
 	for _, kind := range []string{gpucrd.GpuClaimParametersKind} {
 		var err error
 		switch kind {
 		case gpucrd.GpuClaimParametersKind:
 			err = d.gpu.UnsuitableNode(crd, pod, perKindCas[kind], allcas, potentialNode)
+		default:
+			err = fmt.Errorf("unexpected ResourceClaimParameters kind: %+v", kind)
```

Comment:
Can we make this the following to be consistent with line 92:
```
		err = fmt.Errorf("unknown ResourceClaim.ParametersRef.Kind: %v", claim.Spec.ParametersRef.Kind)
```

Reply:
Done.